### PR TITLE
Note on Cabal

### PR DESCRIPTION
**Pull request by Ericson2314**
Oops, I wrote this but forgot to open the PR!

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -302,10 +302,32 @@ dependencies specified in ``.cabal`` files.
 Once GHC becomes multi-target, Stack and cabal-install could use ``-target self``
 to produce the actual program for the compiler host. It would ensure that the
 compiler and ``Setup`` would use the same boot libraries.
+More importantly, it would ensure that `./Setup` can alays be run!
 
 Currently cross-compilers such as GHCJS and Asterius use two GHC compilers: one
 for the target and another for the host (used to build the former GHC, the
 compiler plugins and ``Setup.hs`` programs).
+While with a multi-target GHC we could get away with a single binary, it is traditional for cross-aware build tools to support different tools per platform, and @Ericson2314 thinks this is a good idea.
```
Comment (hsyl20) [inline index.rst]: I don't understand why it would be better to have two binaries

Comment (Ericson2314) [reply index.rst]: Sorry I meant it is good for Cabal to support two multiple binaries. It's definitely preferablecto use one GHC binary where one can get away with it, but I didn't want that to be a reason not to add separate toolchain support to Cabal. Here are two concrete reasons I've thought of:

 - While GHC and Clang are multi-target, other tools like GCC are not so Cabal would already need a nothing of per-stage tools. It's not that much harder to also make that available for GHC itself.

- When using a stage 1 compile that doesn't provide `-target self`, one has the option to instead use the previous stage's compiler to build plugins, which will make the ABI match.

Comment (hsyl20) [reply index.rst]: Ok I see! Thanks, I'll amend the document.

**Comment by Ericson2314** [comment]
Thanks! I guess this one can now be closed too

**Pull request closed without merging**